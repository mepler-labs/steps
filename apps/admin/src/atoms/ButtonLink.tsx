import React from 'react';
import { Link } from 'react-router-dom';
import Button from './Button';

import { remCalc } from 'styles/type';

interface Props {
  className?: string;
  children: any;
  to: string;
  white?: boolean;
}

class ButtonLink extends React.Component<Props, {}> {
  static defaultProps = { white: false };

  render() {
    const { className, children, to, white } = this.props;

    return <Link to={to}><Button {...this.props}>{children}</Button></Link>;
  }
}

export default ButtonLink;
