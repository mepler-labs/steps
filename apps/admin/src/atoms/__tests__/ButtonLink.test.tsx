import ButtonLink from '../ButtonLink';
import { shallow } from 'enzyme';
import * as React from 'react';
import 'jest';

describe('ButtonLink.tsx', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ButtonLink className="my-button-link-class" to="/my/page">
        My button text
      </ButtonLink>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
