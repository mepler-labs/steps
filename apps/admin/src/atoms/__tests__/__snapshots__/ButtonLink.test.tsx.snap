// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink.tsx renders correctly 1`] = `
<Link
  replace={false}
  to="/my/page"
>
  <Button
    className="my-button-link-class"
    to="/my/page"
    white={false}
  >
    My button text
  </Button>
</Link>
`;
